DNN Platform/Modules/DnnExportImport/Components/Services/PortalExportService.cs

```diff
@@ -208,6 +209,15 @@ public override int GetImportTotal()
             var localPortalLanguages =
                 CBO.FillCollection<ExportPortalLanguage>(DataProvider.Instance().GetPortalLanguages(portalId, DateUtils.GetDatabaseUtcTime().AddYears(1), null));
             var localLanguages = CBO.FillCollection<Locale>(DotNetNuke.Data.DataProvider.Instance().GetLanguages());
+
+            // ensure exported languages are enabled in target site
+            foreach(var lang in portalLanguages)
+            {
+
+                LocaleController.Instance.ActivateLanguage(portalId, lang.CultureCode, true);
+                LocaleController.Instance.PublishLanguage(portalId, lang.CultureCode, true);
+            }
+
```

Comment:
I believe that this will fail, if the language isn't actually installed to the target portal.  In the below loop you are going through and looking for local portal support, I would prefer to activate & publish in that loop, after it was validated that the value was created?  Or am I missing something?

Reply:
@mitchelsellers localPortalLanguages is populated with installed languages, so looping through will only activate existing languages.

Reply:
@mean2me, did you test the scenario described by @mitchelsellers? Can you confirm it actually works?

Reply:
@mean2me, if you look in the loop right below your change (which btw is essentially the same loop), you'll notice that the source language is validated to exist in the destination portal:
```c#
var localLanguageId = localLanguages.FirstOrDefault(x => x.Code == exportPortalLanguage.CultureCode)?.LanguageId;
var existingPortalLanguage = localPortalLanguages.FirstOrDefault(t => t.LanguageId == localLanguageId);
if (existingPortalLanguage != null)
{
    // now we know the language exists on the destination portal
}
```
Concern here is that `portalLanguages` is populated with source languages, not destination languages, and thus you should first check if the language exists, and only then activate/publish the language in the destination portal.
Otherwise you're just assuming that the language will exist.
Essentially, remove your loop and move the activation/publication lines to the existing loop, after above validation.
@mitchelsellers correct me if I'm wrong here please.